test/chunking-form/samples/improved-dynamic-chunks/multi-entry-different-dependencies/_expected/es/generated-dynamic.js

```diff
@@ -0,0 +1,6 @@
+import { value1 } from './main1.js';
+export { value1 } from './main1.js';
+import { v as value2 } from './generated-dep2.js';
+export { v as value2 } from './generated-dep2.js';
+
+console.log('dynamic1', value1, value2);
```

Comment:
In this scenario, we have two entry points that both dynamically import a single module. The imported module shares two dependencies with one entry point and only one with the other.

Unfortunately, this is broken: `dep1` should not have been inlined into `main1` because now if the user starts executing the code from the other entry point `main2`, `main1` will also be loaded by the dynamic module. This was not the case before bundling.

Instead, all modules need to become separate chunks
- dep1 cannot be inlined into main1 or main2 because it is shared
- dep2 cannot be inlined into main1 because it is not guaranteed to be in memory when the dynamic chunk is loaded

Reply:
👍 got it thanks for the simpler test. will take a look